pyiron/sphinx/base.py

```diff
@@ -42,45 +42,76 @@
+    scipy.constants.angstrom
 )
 HARTREE_TO_EV = scipy.constants.physical_constants["Hartree energy in eV"][0]
 HARTREE_OVER_BOHR_TO_EV_OVER_ANGSTROM = HARTREE_TO_EV / BOHR_TO_ANGSTROM
 
 
 class SphinxBase(GenericDFTJob):
     """
-    Class to setup and run Sphinx simulations which is a derivative of pyiron_atomistics.job.generic.GenericJob.
-    The functions in these modules are written in such the function names and attributes are very generic
-    (get_structure(), molecular_dynamics(), version) but the functions are written to handle Sphinx specific input and
-    output.
+    Class to setup and run Sphinx simulations.
+
+    Inherits pyiron_atomistics.job.generic.GenericJob. The functions in
+    these modules are written such that the function names and attributes
+    are very Pyiron-generic (get_structure(), molecular_dynamics(),
+    version) but internally handle Sphinx specific input and output.
+
+    Alternatively, because SPHInX inputs are built on a group-based
+    format, users have the option to set specific groups and parameters
+    directly, e.g.
+
+    ```python
+    # Modify/add a new parameter via
+    job.input.hamilton.nEmptyStates = 15
+    job.input.hamilton.dipoleCorrection = True
+    # or
+    job.input.hamilton.set("nEmptyStates", 15)
+    job.input.hamilton.set("dipoleCorrection", True)
+    # Modify/add a sub-group via
+    job.input.guess.rho.charged = {"charge": 2, "z": 25}
+    # or
+    job.input.guess.rho.set("charged", {"charge": 2, "z": 25})
+    ```
 
     Args:
-        project: Project object (defines path where job will be created and stored)
-        job_name (str): name of the job (must be unique within this project path)
+        project: Project object (defines path where job will be
+                 created and stored)
+        job_name (str): name of the job (must be unique within
+                        this project path)
     """
 
     def __init__(self, project, job_name):
         super(SphinxBase, self).__init__(project, job_name)
         self.output = SphinxOutput(job=self)
         self.input = Input()
-        self._main_str = None
-        self._species_str = None
-        self._structure_str = None
-        self._basis_str = None
-        self._hamilston_str = None
-        self._guess_str = None
-        self._spins_str = None
+        self.input.load_default()
         self._save_memory = False
+        self._spin_enabled = False
         self._output_parser = Output(self)
         self.input_writer = InputWriter()
+        self.original_positions = None
+        self.original_species = None
+        self.original_cell = None
         if self.check_vasp_potentials():
             self.input["VaspPot"] = True  # use VASP potentials if available
         self._kpoints_odict = None
         self._generic_input["restart_for_band_structure"] = False
         self._generic_input["path_name"] = None
         self._generic_input["n_path"] = None
 
+    def get_version_float(self):
```

Comment:
Maybe this function can be moved to the executable class in future.

Reply:
created #611